plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/metrics/OTelMetricsTelemetry.java

```diff
@@ -19,19 +20,23 @@
 import io.opentelemetry.api.metrics.DoubleUpDownCounter;
 import io.opentelemetry.api.metrics.Meter;
 import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
 
 /**
  * OTel implementation for {@link MetricsTelemetry}
  */
 public class OTelMetricsTelemetry<T extends MeterProvider & Closeable> implements MetricsTelemetry {
+    private final RefCountedReleasable<OpenTelemetrySdk> refCountedOpenTelemetry;
     private final Meter otelMeter;
     private final T meterProvider;
 
     /**
      * Creates OTel based {@link MetricsTelemetry}.
+     * @param openTelemetry open telemetry.
      * @param meterProvider {@link MeterProvider} instance
      */
-    public OTelMetricsTelemetry(T meterProvider) {
+    public OTelMetricsTelemetry(RefCountedReleasable<OpenTelemetrySdk> openTelemetry, T meterProvider) {
+        this.refCountedOpenTelemetry = openTelemetry;
```

Comment:
Please move ref counter to the class which uses it (and closes it):

```suggestion
        this.refCountedOpenTelemetry = openTelemetry;
        this.refCountedOpenTelemetry.incRef();
```